### PR TITLE
Upgrade to Laravel 5.5

### PR DESCRIPTION
**Pull request by Lavoaster**
~This currently awaits compat support that I added to the poniverse-php library at Poniverse/poniverse-php#3.~

~Once this is done the repository should be removed from the composer.json file, a `composer update` ran to update the lockfile, and a force push to clear that gunk out of our nice history.~ (Minus the force push)

Ready for review.

### CODE REVIEW

File: tests/TestCase.php [hunk 1]
```diff
@@ -46,15 +52,14 @@ public function createApplication()
...
+        $this->getTestFiles();
 
         return $app;
     }
 
-    /**
-     * @before
-     */
-    public function initializeTestFiles()
+    public function getTestFiles()
```
Comment (Lavoaster) [inline tests/TestCase.php]: Just incase this is questioned. PHP Unit 6 ends up calling this prior to the application being booted, which results in no `Storage` facade having been registered - and all tests failing 😢 .

Comment (Feld0) [reply tests/TestCase.php]: 👍

**Comment by Feld0** [comment]
I'm concerned about how those scoped queries worked to begin with. Interesting catch. I'll look at merging this after #123!

File: tests/TestCase.php (hunk 1 above)
Comment (Lavoaster) [reply tests/TestCase.php]: Just realised I probably could have just removed the docblock without renaming, but 🤷‍♀️ it's done now. :P

**Comment by Lavoaster** [comment]
There was a change from Laravel 5.4 to 5.5 where a comparison operator was updated from `?:` to `??`. When meant the falsey value being returned was no longer valid and it wouldn't error correct, it now has to be an explicit null or void return. See below:

```
php > $var = false ?? 'hello world';
php > var_dump($var);
bool(false)
php > $var = false ?: 'hello world';
php > var_dump($var);
string(11) "hello world"
```

Changes: 
5.4: 
https://github.com/laravel/framework/blob/5.4/src/Illuminate/Database/Eloquent/Builder.php#L934

5.5: https://github.com/laravel/framework/blob/5.5/src/Illuminate/Database/Eloquent/Builder.php#L953

**Pull request merged**